src/context/AppContext.js

```diff
@@ -0,0 +1,47 @@
+import React, {useContext, useState} from 'react'
+
+const Context = React.createContext();
+
+export const useApp = () => {
+  return useContext(Context)
+}
+
+const AppProvider = ({children}) => {
+  // global useState
+  const [events, setEvents] = useState([])
+  const [contacts, setContacts] = useState([])
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState('')
+  const [message, setMessage] = useState('')
+
+  // functions
+
+  const getEvents = () => {
```

Comment:
It is also possible that the api call could happen in/around the Apollo provider and be passed through, but we will definitely look into that!